### ISSUE TITLE
[Research] Integrate Apple Health HRV + HR for Multimodal Stress Classification (94.7% accuracy)

### ISSUE THREAD

**Issue by LakshmiSravyaVedantham** [LakshmiSravyaVedantham/DreamAnalysis#22]
## What

Add **Apple Health HRV (LF/HF ratio) and Heart Rate** as features to the stress classifier alongside the existing EEG Theta/Alpha Ratio (TAR). A Feb 2026 Scientific Reports paper shows this specific 3-feature combination achieves **94.7% stress classification accuracy** with a simple SVM.

## Why

- This is the **lowest-effort, highest-impact** multimodal fusion improvement
- Current stress classifier uses EEG-only features (~65-75% accuracy)
- Adding just 2 features from Apple Health (HR + LF/HF ratio from HRV) → **94.7% accuracy**
- NeuralDreamWorkshop already computes TAR (Theta/Alpha Ratio) from Muse 2
- Apple Health integration already exists in `ml/health/` directory
- Gender-specific finding: females show better classification — add gender as feature

### Step 1: Extract HRV features from Apple Health
```python
# ml/health/hrv_features.py
def extract_stress_features_from_health(health_data: dict) -> dict:
    """Extract HR and HRV features from Apple Health snapshot.
    
    Returns:
        heart_rate: Current or most recent HR (bpm)
        lf_hf_ratio: Low-frequency / High-frequency HRV ratio
            LF (0.04-0.15 Hz) = sympathetic + parasympathetic
            HF (0.15-0.40 Hz) = parasympathetic only
            High LF/HF = stress/sympathetic dominance
    """
```

### Step 2: Update stress classifier
```python
# In ml/models/stress_detector.py or emotion_classifier.py
def predict_stress_multimodal(eeg_features, health_features):
    """3-feature SVM stress classifier.
    
    Features:
    1. TAR (Theta/Alpha Ratio) — from EEG (already computed)
    2. Heart Rate — from Apple Health
    3. LF/HF Ratio — from Apple Health HRV
    
    Optional: add gender as 4th feature for +2% accuracy
    """
    from sklearn.svm import SVC
    # Load pretrained SVM or train on collected data
```

### Step 3: API integration
- Update `/analyze-eeg` to optionally accept health_data parameter
- If Apple Health data available, use multimodal stress classifier
- If not, fall back to EEG-only stress index (current behavior)
- Return `stress_model: "multimodal_svm" | "eeg_heuristic"` in response

### Step 4: Frontend
- On EEG monitor page, show "Enhanced stress detection" badge when health data is available
- Pull latest HR and HRV from Apple Health before/during EEG session

### Files to Modify
- `ml/health/hrv_features.py` (NEW) — HRV feature extraction
- `ml/models/stress_detector.py` — Add multimodal SVM path
- `ml/models/emotion_classifier.py` — Wire multimodal stress into emotion output
- `ml/api/routes.py` — Accept health_data in `/analyze-eeg`
- `client/src/pages/eeg-monitor.tsx` — Show enhanced detection indicator

### Expected Impact
- Stress classification: **65-75% → 92-95%** with Apple Health data
- Essentially solves stress detection with 3 simple features

## Complexity
LOW — SVM on 3 features, Apple Health integration already exists

## References
- EEG-ECG multimodal stress (Scientific Reports Feb 2026): https://www.nature.com/articles/s41598-026-38356-3
- Features validated: TAR (EEG), HR, LF/HF ratio (HRV)
- Best model: SVM, 94.7% combined-gender, 92.6% avg across conditions
